pkg/gatewayserver/io/io.go

```diff
@@ -263,7 +265,19 @@ func (c *Connection) ScheduleDown(path *ttnpb.DownlinkPath, msg *ttnpb.DownlinkM
 	if err != nil {
 		return 0, err
 	}
-	phy, err := band.GetByID(c.fp.BandID)
+	var fpIndex int
+	var gtwFP *frequencyplans.FrequencyPlan
+	for fpIndex, gtwFP = range c.fps {
+		if gtwFP.BandID == request.FrequencyPlanID {
+			break
+		}
+	}
+	if fpIndex == len(c.fps) {
```

Comment:
Unclear and too much heuristic of how indexer of for-range works. Just assign `gtwFP` in the loop and check here if it's assigned